usr/bs_hyc.c

```diff
@@ -118,13 +115,32 @@ static int bs_hyc_cmd_submit(struct scsi_cmd *cmdp)
 	assert(infop->rpc_con != kInvalidRpcHandle);
 
 	op = scsi_cmd_operation(cmdp);
-	if (hyc_unlikely(op == UNKNOWN)) {
-		return -EINVAL;
+	if (hyc_unlikely(op == WRITE_SAME_OP)) {
+		return -1;
+	} else if (hyc_unlikely(op == UNKNOWN)) {
+		return 0;
 	}
 
 	offset = scsi_cmd_offset(cmdp);
 	length = scsi_cmd_length(cmdp);
 
+	/*
```

Comment:
Need to test this case of zero size IO. Check how to issue barrier IO (check sg_sync scsi command) which are genrally of zero size. We can also try to test using dd by giving it a zero size file for reading (simply touch the file and then try to write it) 

touch /tmp/file1
dd if=/tmp/file1 of=/dev/<discoved_device> count=1 bs=4096 oflag=direct 

If dd is not working then we have to write our own client which opens the device and issue a zero length IO.